nanofi/include/sitetosite/CPeer.h

```diff
@@ -74,8 +74,8 @@ static void setHostName(struct SiteToSiteCPeer * peer, const char * hostname) {
   peer->_url = (char*)malloc(host_len + 14);  // +1 for trailing zero, 1 for ':', at most 5 for port, 7 for "nifi://" suffix
   memset(peer->_url, 0, host_len + 14);  // make sure to have zero padding no matter the length of the port
   strncpy(peer->_host, hostname, host_len);
-  strncpy(peer->_url, "nifi://", 7);
-  strncpy(peer->_url + 7, hostname, host_len);
+  memcpy(peer->_url, "nifi://", 7);
```

Comment:
Do you not just mask out the reported mistake of not copying `\0`?
```bash
nifi://
0123456 --> 7 characters copied, missing \0.
```

Reply:
Updated in 828f564566b9e7fa50bfe0f37bcbf0996b22de67